### DISCUSSION TITLE
New feature "Preview Changes" slows down updates

### DISCUSSION THREAD

**Discussion by arlem03** [dbeaver/dbeaver#35773] [Q&A] [answered]
Hello. Thank you for this awesome database management software. Recently, in the last update, a new function "Preview Changes" was added. This makes updates much slower, especially when one has to make thousands of updates. Having to click on the save button with the mouse every single time is annoying and not needed at all. 

Is there a way to make this "Preview Changes" in the SQL Editor optional or can we remote it all together?

It doesn't seem much but for those who do a lot of database changes, it's a pain to have to handle the prompt every time, and also to have to get the hands of the keyboards to handle the mouse.

**Comment by E1izabeth** [comment]
I appreciate your request. Could you tell me if you're talking about this dialog?
![image](https://github.com/user-attachments/assets/0ea95f56-026e-42b1-b579-677ce4003b68)

Could you provide me with steps to reproduce the behavior when you get this dialog unintentionally? It doesn't show up for me when I press Ctrl+S or the Save button.

**Comment by arlem03** [comment]
Hello Elizabeth, 

Thank you for looking into this: 

The Dialog looks like this: 

The steps are two folds. 

1. I do a query from one of my databases. 
2. I change a couple of values, or part of values. 
3. I either do a CTRL-S or press the button "Save." 
4. I get the above "Preview Changes" prompt, which requires me to confirm the save using the mouse. Best would be to work as it used to be--meaning without this new prompt. Could be an option to have it either way. 

As a note, I use the Linux version 24.2.0.

**Comment by E1izabeth** [comment]
Unfortunately, GitHub can't attach files when you answer by email, so I don't see the files you sent. Can you please go to GitHub and attach them there?

**Comment by arlem03** [comment]
Got it. Here is the snapshot of the prompt.
![Preview_changes](https://github.com/user-attachments/assets/8f69b7bd-7850-446e-87d0-23d7d808fa1b)

**Comment by arlem03** [comment]
No problem. I just did this.

**Comment by E1izabeth** [comment, marked as answer]
You can change it in the Connection type properties. This dialog can be switched off by `Confirm data changes` setting
![image](https://github.com/user-attachments/assets/5c676b50-d01f-4ac7-b430-650b0fa01446)

**Comment by arlem03** [comment]
Hello Elizabeth, 

What a relief!!! Thank you so much for you help. I didn't know that changing to "production" would change the prompts. This is very good news and totally resolved the issue I was having. You're the best! 

Thanks, 
Robert
